samples/net/mqtt_sn_publisher/src/main.c

```diff
@@ -107,9 +114,10 @@ int main(void)
 	k_thread_access_grant(k_current_get(), &run_app);
 	k_mem_domain_add_thread(&app_domain, k_current_get());
 
-	k_thread_user_mode_enter((k_thread_entry_t)start_client, NULL, NULL, NULL);
+	k_thread_user_mode_enter(start_client, &result, NULL, NULL);
```

Comment:
I may be wrong here, but I am suspicious of passing _&result_ for the user thread case as the memory to which it points may not be available to the user thread. If this suspicion is correct, then that would result in an exception.

My recommendation would be to pass _NULL_ for the user thread case and have a check in _start_client()_ to only write to _*result_ if _result_ is not _NULL_.

Reply:
I would rather skip the result completely, as it is only used for exit() later on. I don't really see the benefit of reporting something there.

Reply:
Yeah, this would definitely result in MPU / MMU exception.

Reply:
I removed the result completely, as I do not think it is really helpful. This way now the sample has the same exit code, no matter if CONFIG_USERSPACE is enabled or not.